R/cohenFleissKappa.R

```diff
@@ -0,0 +1,202 @@
+  if(options[["cohensKappa"]])
+    jaspResults[["cohensKappa"]] <- .cohensKappa(dataset, options, ready)
+  
+  if(options[["fleissKappa"]])
+    jaspResults[["fleissKappa"]] <- .fleissKappa(dataset, options, ready)
+  
+  return()
+}
+
+# Read in the dataset (copied from .reliabilityReadData)
+.cohensFleissKappaReadData <- function(dataset, options) {
+  variables <- unlist(options[["variables"]])
+  if (is.null(dataset)) {
+    dataset <- .readDataSetToEnd(columns.as.factor = variables)
+  }
+  return(dataset)
+}
+
+.cohensKappa <- function(dataset, options, ready){
+  
+  weighted <- options[["cohensWeightedOrNot"]] == "cohensWeighted"
+  
+  if(weighted){
+    weightedString <- "Weighted"
+  }else{
+    weightedString <- "Unweighted"
+  }
+  
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Cohen's %s Kappa", casefold(weightedString, upper = F)))
+  jaspTable$addColumnInfo(name = "ratings", title = gettext("Ratings"), type = "string")
+  jaspTable$addColumnInfo(name = "cKappa", title = gettextf("%s Kappa", weightedString), type = "number")
+  jaspTable$position <- 1
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "cohensKappa",
+      "cohensWeightedOrNot",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+  
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  
+  if(ready){
+    #Create all pairs
+    variables <- colnames(dataset)
+    allPairs <- .getPairs(variables)
+    allDataframes <- .getDataframes(dataset, allPairs)
+    allPairStrings <- paste(allPairs$x, allPairs$y, sep = " - ")
+    
+    #calculate Cohen's Kappas
+    allKappaData <- lapply(allDataframes, psych::cohen.kappa, alpha = 1 - options[["kappaConfidenceIntervalValue"]])
+    
+    #Extract Kappas and CIs
+    allKappas <- c()
+    allLowerBounds <- c()
+    allUpperBounds <- c()
+    
+    for(i in allKappaData){
+      kappaData <- i$confid
+      if(weighted){
+        k <- 2
+      }else{
+        k <-1
+      }
+      allKappas <- c(allKappas, kappaData[k,2])
+      allLowerBounds <- c(allLowerBounds, kappaData[k,1])
+      allUpperBounds <- c(allUpperBounds, kappaData[k,3])
+    }
+    
+    tableData <- list("ratings" = allPairStrings,
+                      "cKappa" = allKappas)
+    
+    if (options[["kappaIntervalOn"]]) {
+      jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      tableData[["CIL"]] <- allLowerBounds
+      tableData[["CIU"]] <- allUpperBounds
+    }
+    
+    
+    jaspTable$setData(tableData)
+    jaspTable$addFootnote(gettextf('%i subjects/items and %i judges/measurements.', nrow(dataset), ncol(dataset)))
```

Comment:
Yes, I agree. @jansim  maybe we should also change this in the ICC for consistency?

Reply:
Yeah, good call! @JTPetter  I'm also mainly using 'rater' in the ICC Helpfile so it should work well 👍 

Do you guys have a tip on how to go about this? As the string is already translated, should I just remove translations for that string? It does seem like kind of a waste, though, since only one word changed 🤔

Reply:
I guess I also missed the "judges" in the ICC file. Sorry. @jansim What do you mean by translations? You mean translations of the helpfile? For the translations of the tables and stuff, I would just leave that to the translators. I dont think it is much of a concern for us.

Reply:
Ok, should I remove the obsolete lines from the .po files then or will that be detected automatically and I'll only adjust the R code?

Reply:
I think you can leave the .po files alone and only adjust the R-code, but @boutinb might know the usual procedure here.